[BUG] Training fails due to fastai ImportError when already installed

**Bug Report Checklist**
<!-- Please ensure at least one of the following to help the developers troubleshoot the problem:  -->
- [ x ] I provided code that demonstrates a minimal reproducible example. <!-- Ideal, especially via source install --> 
- [ ] I confirmed bug exists on the latest mainline of AutoGluon via source install. <!-- Preferred --> 
- [ x ] I confirmed bug exists on the latest stable version of AutoGluon. <!-- Unnecessary if prior items are checked --> 

**System**
- Windows 11
- Threadripper 9955wx
- 128GB RAM
- 2x RTX 5090 (only 1 used during training)

**Describe the bug**
NeuralNetFastAI models fail during training due to faulty ImportError

**Expected behavior**
Models should train successfully

**To Reproduce**
[Code here](https://gist.github.com/RobertAgee/9c1f1e0afb8bd332e4ef6d50de708863)

**Screenshots / Logs**
```
Fitting 2 child models (S1F1 - S1F2) | Fitting with ParallelLocalFoldFittingStrategy (2 workers, per: cpus=16, gpus=0, memory=0.05%)
Warning: Exception caused NeuralNetFastAI_r102_BAG_L1 to fail during training (ImportError)... Skipping this model.
Import fastai failed. A quick tip is to install via `pip install autogluon.tabular[fastai]==1.5.0`.
```

<img width="1105" height="139" alt="Image" src="https://github.com/user-attachments/assets/6326ce29-51da-40aa-b063-d7c6e641686f" />

<img width="1104" height="101" alt="Image" src="https://github.com/user-attachments/assets/dbe6f275-02c9-4e87-9ad9-2e21ab932be5" />

**Installed Versions**
<img width="973" height="195" alt="Image" src="https://github.com/user-attachments/assets/a68e87cf-8ca5-45a3-b311-ef00996e5e7a" />

<img width="1087" height="216" alt="Image" src="https://github.com/user-attachments/assets/083f2dd2-516e-45e2-ac31-f49fceba5e48" />

```
date                   : 2026-01-03
time                   : 04:20:27.591180
python                 : 3.10.19.final.0
OS                     : Windows
OS-release             : 10
Version                : 10.0.26100
machine                : AMD64
processor              : AMD64 Family 26 Model 8 Stepping 1, AuthenticAMD
num_cores              : 32
cpu_ram_mb             : 130408.234375
cuda version           : None
num_gpus               : 2
gpu_ram_mb             : [32187, 30425]
avail_disk_size_mb     : None

accelerate             : 1.12.0
autogluon              : 1.5.0
autogluon.common       : 1.5.0
autogluon.core         : 1.5.0
autogluon.features     : 1.5.0
autogluon.multimodal   : 1.5.0
autogluon.tabular      : 1.5.0
autogluon.timeseries   : 1.5.0
boto3                  : 1.42.21
catboost               : 1.2.8
chronos-forecasting    : 2.2.2
coreforecast           : 0.0.16
defusedxml             : 0.7.1
einops                 : 0.8.1
einx                   : 0.3.0
evaluate               : 0.4.6
fastai                 : 2.8.6
fsspec                 : 2025.3.0
fugue                  : 0.9.4
gluonts                : 0.16.2
huggingface_hub        : 0.36.0
hyperopt               : 0.2.7
imodels                : None
interpret-core         : None
jinja2                 : 3.1.6
joblib                 : 1.5.3
jsonschema             : 4.23.0
lightgbm               : 4.6.0
lightning              : 2.5.6
loguru                 : 0.7.3
matplotlib             : 3.10.8
mlforecast             : 0.14.0
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.9.2
numpy                  : 2.1.3
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.3.0
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
orjson                 : 3.11.5
pandas                 : 2.3.3
pdf2image              : 1.17.0
peft                   : 0.17.1
Pillow                 : 11.3.0
psutil                 : 7.1.3
pyarrow                : 20.0.0
pytabkit               : None
pytesseract            : 0.3.13
pytorch-metric-learning: 2.8.1
pyyaml                 : 6.0.3
ray                    : 2.52.1
requests               : 2.32.5
scikit-image           : 0.25.2
scikit-learn           : 1.7.2
scikit-learn-intelex   : None
scipy                  : 1.15.3
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.8.11
statsforecast          : 2.0.1
stevedore              : 5.4.1
tabdpt                 : None
tabicl                 : None
tabpfn                 : None
tensorboard            : 2.20.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.9.1+cu128
torchmetrics           : 1.7.4
torchvision            : 0.24.1+cu128
tqdm                   : 4.67.1
transformers           : 4.57.3
utilsforecast          : 0.2.11
xgboost                : 3.1.2
```

Also, you can see it incorrectly reports CUDA is not found but it is installed with torch

Looks like a related (same?) issue broke our CI due to a recent `fastai-2.8.6` release: https://github.com/autogluon/autogluon/actions/runs/20698278682/job/59554171252

To reproduce
```bash
pip install autogluon==1.5 pytest
pytest tabular/unittests/models/test_tabular_nn_fastai.py
```
Fails with
```
ImportError while importing test module '/home/shchuro/workspace/autogluon/tabular/tests/unittests/models/test_tabular_nn_fastai.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../../.local/share/uv/python/cpython-3.12.9-linux-x86_64-gnu/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
unittests/models/test_tabular_nn_fastai.py:5: in <module>
    from fastai.callback.core import CancelFitException
../../../../envs/test/lib/python3.12/site-packages/fastai/callback/core.py:7: in <module>
    from ..data.all import *
../../../../envs/test/lib/python3.12/site-packages/fastai/data/all.py:1: in <module>
    from ..torch_basics import *
../../../../envs/test/lib/python3.12/site-packages/fastai/torch_basics.py:9: in <module>
    from .imports import *
../../../../envs/test/lib/python3.12/site-packages/fastai/imports.py:34: in <module>
    from fastprogress.fastprogress import progress_bar,master_bar
../../../../envs/test/lib/python3.12/site-packages/fastprogress/__init__.py:2: in <module>
    from .fastprogress import master_bar, progress_bar, force_console_behavior
../../../../envs/test/lib/python3.12/site-packages/fastprogress/fastprogress.py:17: in <module>
    from IPython.display import display,HTML,Markdown
E   ModuleNotFoundError: No module named 'IPython'
```

I suspect that this change https://github.com/fastai/fastai/commit/ad8e3744158d9fce98e48361afc0d9c0d6f4f3a2 skips the `IPython` installation when installing `fastai`, which breaks the import of `NeuralNetFastAI`.

@RobertAgee can you try installing `pip install ipython` and see if that fixes the problem?

@Innixma should we hotfix this in v1.5.1?

--- a/tabular/setup.py
+++ b/tabular/setup.py
@@ -55,7 +55,7 @@ extras_require = {
     "fastai": [
         "spacy<3.9",
         "torch",  # version range defined in `core/_setup_utils.py`
-        "fastai>=2.3.1,<2.9",  # <{N+1} upper cap, where N is the latest released minor version
+        "fastai>=2.3.1,<2.8.6",  # Cap due to dependency conflict in fastai-2.8.6 https://github.com/autogluon/autogluon/issues/5521
     ],
     "tabm": [
         "torch",  # version range defined in `core/_setup_utils.py`

